src/plugins/discover/public/plugin.ts

```diff
@@ -257,7 +256,6 @@ export class DiscoverPlugin
         }
         setScopedHistory(params.history);
         setHeaderActionMenuMounter(params.setHeaderActionMenu);
-        syncHistoryLocations();
```

Comment:
I think this is no longer need as the current fix should cover this. 
This was added in the scope "explore underlying data" action where we navigate from dashboard to discover. Worth testing "explore underlying data" when testing this pr